src/documents/serialisers.py

```diff
@@ -681,12 +686,48 @@ def update(self, instance: Document, validated_data):
                             custom_field_instance.field,
                             doc_id,
                         )
+        if (
+            "remove_inbox_tags" in validated_data
+            and validated_data["remove_inbox_tags"]
+        ):
+            tag_ids_being_added = (
+                [
+                    tag.id
+                    for tag in validated_data["tags"]
+                    if tag not in instance.tags.all()
+                ]
+                if "tags" in validated_data
+                else []
+            )
+            inbox_tags_not_being_added = Tag.objects.filter(is_inbox_tag=True).exclude(
+                id__in=tag_ids_being_added,
+            )
+            if "tags" in validated_data:
+                validated_data["tags"] = [
+                    tag
+                    for tag in validated_data["tags"]
+                    if tag not in inbox_tags_not_being_added
+                ]
+            else:
+                validated_data["tags"] = [
+                    tag
+                    for tag in instance.tags.all()
+                    if tag not in inbox_tags_not_being_added
+                ]
         super().update(instance, validated_data)
         return instance
 
     def __init__(self, *args, **kwargs):
         self.truncate_content = kwargs.pop("truncate_content", False)
 
+        # return full permissions if we're doing a PATCH or PUT
+        context = kwargs.get("context")
+        if (
+            context.get("request").method == "PATCH"
+            or context.get("request").method == "PUT"
+        ):
+            kwargs.__setitem__("full_perms", True)
```

Comment:
Does `kwargs["full_perms"] = True` not work?  Seems odd to use a magic method here

Reply:
Im not even really sure how I ended up with that one, but indeed this works fine. Thanks!